src/components/modes/teacher/TeacherView.js

```diff
@@ -1,282 +1,85 @@
 import SettingsIcon from '@material-ui/icons/Settings';
-import TableRow from '@material-ui/core/TableRow';
-import Typography from '@material-ui/core/Typography';
-import Paper from '@material-ui/core/Paper';
-import Button from '@material-ui/core/Button';
-import IconButton from '@material-ui/core/IconButton';
-import DeleteIcon from '@material-ui/icons/Delete';
-import RefreshIcon from '@material-ui/icons/Refresh';
-import Select from 'react-select';
 import { withTranslation } from 'react-i18next';
+import { Paper } from '@material-ui/core';
 import PropTypes from 'prop-types';
-import './TeacherView.css';
-import {
-  patchAppInstanceResource,
-  postAppInstanceResource,
-  deleteAppInstanceResource,
-  openSettings,
-} from '../../../actions';
-import { getUsers } from '../../../actions/users';
-import { addQueryParamsToUrl } from '../../../utils/url';
+import { toggleSettings } from '../../../actions';
 import Settings from './Settings';
+import Main from '../../common/Main';
+import { addQueryParamsToUrl } from '../../../utils/url';
+import { MODES } from '../../../config/settings';
 
-/**
- * helper method to render the rows of the app instance resource table
- * @param appInstanceResources
- * @param dispatchPatchAppInstanceResource
- * @param dispatchDeleteAppInstanceResource
- * @returns {*}
- */
-const renderAppInstanceResources = (
-  appInstanceResources,
-  { dispatchPatchAppInstanceResource, dispatchDeleteAppInstanceResource },
-) => {
-  // if there are no resources, show an empty table
-  if (!appInstanceResources.length) {
-    return (
-      <TableRow>
-        <TableCell colSpan={4}>No App Instance Resources</TableCell>
-      </TableRow>
-    );
-  }
-  // map each app instance resource to a row in the table
-  return appInstanceResources.map(({ _id, appInstance, data }) => (
-    <TableRow key={_id}>
-      <TableCell scope="row">{_id}</TableCell>
-      <TableCell>{appInstance}</TableCell>
-      <TableCell>{data.value}</TableCell>
-      <TableCell>
-        <IconButton
-          color="primary"
-          onClick={() => {
-            dispatchPatchAppInstanceResource({
-              id: _id,
-              data: { value: Math.random() },
-            });
-          }}
-        >
-          <RefreshIcon />
-        </IconButton>
-        <IconButton
-          color="primary"
-          onClick={() => dispatchDeleteAppInstanceResource(_id)}
-        >
-          <DeleteIcon />
-        </IconButton>
-      </TableCell>
-    </TableRow>
-  ));
-};
-
-const generateRandomAppInstanceResource = ({
-  dispatchPostAppInstanceResource,
-}) => {
-  dispatchPostAppInstanceResource({
-    data: { value: Math.random() },
-  });
-};
+const styles = (theme) => ({
+  message: {
+    margin: theme.spacing(2),
+    padding: theme.spacing(1),
+    textAlign: 'center',
+    backgroundColor: theme.status.danger.background[500],
+    color: theme.status.danger.color,
+  },
+  fab: {
+    zIndex: 10000,
```

Comment:
Can we factor this out like above? Not sure what to call it, though. `FAB_Z_INDEX`?